### PR TITLE
feat: Add ability to use custom domains

### PR DESCRIPTION
**Pull request by foadlind**
* Set a custom domain for S3 using the `S3_CUSTOM_DOMAIN` config
  parameter.
* Use `S3_PROFILE_IMAGE_CUSTOM_DOMAIN` to set a separate custom domain
  for profile images.

### CODE REVIEW

**Review by fghaas — COMMENTED** [review]
See comments inline; my approach would be to only set those values if the corresponding config option is set.

File: tutors3/plugin.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
         "PROFILE_IMAGE_MAX_AGE": "31536000",
         "ADDRESSING_STYLE": "auto",
         "SIGNATURE_VERSION": "s3v4",
+        "CUSTOM_DOMAIN": "",
```
[the remark this answers is not in the document]
Comment (foadlind) [reply tutors3/plugin.py]: Yes that makes sense

File: tutors3/patches/openedx-lms-common-settings [hunk 2]
```diff
@@ -3,5 +3,6 @@ PROFILE_IMAGE_BACKEND = {
     "options": {
         "bucket": "{{ S3_PROFILE_IMAGE_BUCKET }}",
         "location": PROFILE_IMAGE_BACKEND["options"]["location"].lstrip("/"),
+        "custom_domain": "{{ S3_PROFILE_IMAGE_CUSTOM_DOMAIN }}",
```
[the remark this answers is not in the document]
Comment (foadlind) [reply tutors3/patches/openedx-lms-common-settings]: Will do

File: tutors3/patches/openedx-auth [hunk 3]
```diff
@@ -1,3 +1,3 @@
 "AWS_STORAGE_BUCKET_NAME": "{{ S3_STORAGE_BUCKET }}",
-"AWS_S3_CUSTOM_DOMAIN": "",
+"AWS_S3_CUSTOM_DOMAIN": "{{ S3_CUSTOM_DOMAIN }}",
```
[the remark this answers is not in the document]
Comment (foadlind) [reply tutors3/patches/openedx-auth]: will do

**Review by fghaas — COMMENTED** [review]
Looks good now, functionally. I'd suggest we should also add a note in the README saying that due to a limitation in Open edX, if you're using `s3v4` signatures, your `S3_PROFILE_IMAGE_BUCKET` must have a public ACL and you must set `S3_PROFILE_IMAGE_CUSTOM_DOMAIN`. Does that sound reasonable?

**Comment by foadlind** [comment]
Yes, I added the note and also updated the commit message.

**Review by fghaas — APPROVED** [review]
Looks good to me now, thank you!

**Pull request merged**